tailcfg/tailcfg.go

```diff
@@ -830,6 +832,26 @@ type Hostinfo struct {
 	//       require changes to Hostinfo.Equal.
 }
 
+// VIPService represents a service created on a tailnet from the
+// perspective of a node providing that service. These services
+// have an IP address pair distinct from the node's IPs.
+type VIPService struct {
+	// Ports specify which ProtoPorts are made available by this node
+	// on the service's IPs.
+	Ports []ProtoPortRange
+
+	// Name is the name of the service, of the form `svc:dns-label`.
+	// See CheckServiceName for a validation func.
+	// Name uniquely identifies a service on a particular tailnet,
+	// and so also corresponds uniquely to the pair of IP addresses
+	// belonging to the VIP service.
+	Name string
+
+	// Active specifies whether new requests for the service should be
+	// sent to this node by control.
+	Active bool
```

Comment:
Question: what is the intention of this field? In what cases/by whom would it be set to false?

Reply:
This is the inverse of the `Draining` field we talked about, it's just named more generally. It'll be false when you have a valid serve config for a service but stop advertising it

Reply:
> stop advertising it

For a 'draining'/pre-shutdown behaviour I think the key for us would be that any existing connections can continue till the peer at the other end has received a netmap update with another preferred endpoint for that service; I don't have a mental model for whether 'stops advertising it' would allow for this

Reply:
Resolved this with @irbekrm offline